pkg/internal/vault/vault_test.go

```diff
@@ -108,17 +183,29 @@ type testSignT struct {
 	expectedCA   string
 }
 
-func TestSign(t *testing.T) {
-	privatekey := generateRSAPrivateKey(t)
-	csrPEM := generateCSR(t, privatekey)
-
+func bundlePEM(csrPEM []byte, caPEM ...string) ([]byte, error) {
 	bundle := certutil.Secret{
 		Data: map[string]interface{}{
 			"certificate": testCertBundle,
 		},
 	}
+	if len(caPEM) > 0 {
+		bundle.Data["ca_chain"] = caPEM
+	}
+	return jsonutil.EncodeJSON(&bundle)
+}
+
+func TestSign(t *testing.T) {
+	privatekey := generateRSAPrivateKey(t)
+	csrPEM := generateCSR(t, privatekey)
+
+	bundleData, err := bundlePEM(csrPEM, testIntermediateCa)
```

Comment:
i don't think a certificate can be distributed only with the intermediate CA, is there a step to verify if a certificate is valid? pretty sure it would error then:

validating locally with `openssl` (configuration basically identically as the one you provided):

```shell
# without certificate chain, only intermediate CA
openssl verify -CAfile openssl/intermediate/intermediate.cert.pem openssl/example.com.cert.pem 
openssl/example.com.cert.pem: C = NA, ST = Nowhere, O = Anonymous, OU = NA, CN = Anonymous Intermediate CA, emailAddress = user@example.com
error 2 at 1 depth lookup:unable to get issuer certificate

# with certificate chain
$ openssl verify -CAfile openssl/intermediate/chain.cert.pem openssl/example.com.cert.pem
openssl/example.com.cert.pem: OK
```

so, unless it is intended to test this process, working only with the whole certificate chain would do the job.

Reply:
the intent was to to include a test with one CA. I'll use a better example

Reply:
> i don't think a certificate can be distributed only with the intermediate C

many (especially ACME) CAs do this as the root cert would already be on the server's trust chain

Reply:
Wouldn't the user then use an ACME issuer configuration instead of Vault then? I opted for Vault exactly because i'm using self-signed CAs

Reply:
Also, as @crandles stated on [his comment](https://github.com/jetstack/cert-manager/pull/3340#issuecomment-702829316), these certificates were indeed generated by a self-signed Root CA and thus this test would fail on a real validity test on the leaf certificate, as i stated on the beginning of this thread.

In my opinion, @crandles used the right process to generate certificates and just the test is lacking a `verify` step, like one would on terminal with `openssl`, which would fail if the changes i've recommended is not implemented somehow.

Reply:
An example on how to verify: https://golang.org/pkg/crypto/x509/#example_Certificate_Verify

Reply:
Going on, i was wrong when i stated `which would fail if the changes i've recommended are not implemented somehow`, it will not fail, both `openssl verify` and Istio responded fine to CA certificates chain being repeated on both `ca.crt` and `tls.crt` , but why do so? I don't think cert-manager should be concerned on how a specific application will use the certificates but actually with how certificate managers (automated or not) will be provided with certificates and application owners use certificates as they will. On the other hand it could provide an easy way to have the whole bundle to be used, as i mentioned [here](https://github.com/jetstack/cert-manager/pull/3340#discussion_r501272609).

If you fellas decide to go trough with this, it should be created as feature request or something of a sort, the changes @crandles made will work just fine with the current design of certificates providing, expect the verification with only the intermediate CA, as it was generated with a self-signed Root CA.